### ISSUE TITLE
Optional spring security dependencies lead to ClassNotFoundException

### ISSUE THREAD

**Issue by michbeck100** [zkoss/zkspring#30]
Why are these dependencies optional? They are obviously needed for compilation. 
```
<dependency>
	<groupId>org.springframework.security</groupId>
	<artifactId>spring-security-config</artifactId>
	<version>${springsecurity.version}</version>
	<optional>true</optional>
</dependency>
<dependency>
	<groupId>org.springframework.security</groupId>
	<artifactId>spring-security-taglibs</artifactId>
	<version>${springsecurity.version}</version>
	<optional>true</optional>
</dependency>
```
The `optional` tag is used if a dependency is really optional, so developers can choose which implementation they use (like a jdbc driver), but it's not needed at runtime. These dependencies are definitely needed. Unfortunately this leads to this error: https://forum.zkoss.org/question/92707/javalangclassnotfoundexception-orgspringframeworksecurityaclsmodelsidretrievalstrategy/

**Comment by cor3000** [comment]
yes we made it optional to 'force' the user to pick their dependency deliberately.
Since everyone is picking their favorite version the one we pre-pick is most likely the wrong one. In the best case just causing version conflicts, in the worst case it's out-dated introducing unexpected security vulnerabilities.

Our examples show how to specify your own version. E.g. [here in this repo ](https://github.com/zkoss/zkspring/blob/4.0.0/zkspringessentials/zkspringcoresec/pom.xml#L76-L90)
Or the [spring-boot way](https://github.com/zkoss/zkspringboot/blob/v2.3.0/zkspringboot-demos/zkspringboot-security-demo/pom.xml#L34-L38).

Especially when it comes to security I prefer a broken build over the wrong version picked from a transitive dependency.

**Comment by michbeck100** [comment]
Not sure if I understand you correct, but if I use a different spring version in my project, zkspring will automatically pick this version instead of the original spring version in zkspring. So this is not a problem. The bigger problem is, that `spring-security-config` and `spring-security-taglibs` are completely missing in the classpath if I don't add it accidentally dently to my projects `pom.xml`.

**Comment by cor3000** [comment]
Security concerns are not added "accidentally" to a project.
My point is even if this is not initially intuitive, there should be an active decision to add a specific spring security version. If it just happened to work out of the box you'll likely not get the best version for your project. In addition your spring security config is usually runnable/testable without zkspring. So you'll have this in place before adding a ZK integration layer.

**Comment by cor3000** [comment]
**TODO**: post FR to 
* improve error handling ... inform about missing dependencies more gracefully at startup
* ensure the taglib functions don't fail if an unused optional dependency is missing

**Comment by cor3000** [comment]
posted FR -> 
[ZKSPRING-59](https://tracker.zkoss.org/browse/ZKSPRING-59) - Enhance optional spring-security dependency handling

**Comment by cor3000** [comment]
implemented and available nightly 4.0.1-FL-2020062213 (for testing)

**Comment by cor3000** [comment]
v4.0.1 available on maven
